api/wasm/indigo-ketcher/indigo-ketcher.cpp

```diff
@@ -503,44 +501,50 @@ namespace indigo
         }
         int base = 0;
-        calculate_iteration_object(IndigoObject(_checkResult(indigoIterateReactants(iko.id()))), molecularWeightStream, mostAbundantMassStream,
-                                   monoisotopicMassStream, massCompositionStream, grossFormulaStream, selected_atoms, base);
-
-        std::stringstream mws, mams, misos, mcs, gfs;
-        calculate_iteration_object(IndigoObject(_checkResult(indigoIterateProducts(iko.id()))), mws, mams, misos, mcs, gfs, selected_atoms, base);
-        if (grossFormulaStream.str().size() && gfs.str().size()) // check if we have gross formula for both reagents and products
-        {
+        calculate_iteration_object(IndigoObject(_checkResult(indigoIterateReactants(iko.id()))), molWeights[0], mamMasses[0], misoMasses[0], massCompositions[0], grossFormulas[0], selected_atoms, base);
+        calculate_iteration_object(IndigoObject(_checkResult(indigoIterateProducts(iko.id()))), molWeights[1], mamMasses[1], misoMasses[1], massCompositions[1], grossFormulas[1], selected_atoms, base);
+        
+        if (grossFormulas[0].size() && grossFormulas[1].size()) // check if we have gross formula for both reagents and products
+        {
+            molecularWeightStream << VectorStringBracketed(molWeights[0], true ); // true means we have both: products and reagents so all components are bracketed
+            mostAbundantMassStream << VectorStringBracketed(mamMasses[0], true );
+            monoisotopicMassStream << VectorStringBracketed(misoMasses[0], true );
+            massCompositionStream << VectorStringBracketed(massCompositions[0], true );
+            grossFormulaStream << VectorStringBracketed(grossFormulas[0], true );
             molecularWeightStream << " > ";
             mostAbundantMassStream << " > ";
             monoisotopicMassStream << " > ";
             massCompositionStream << " > ";
             grossFormulaStream << " > ";
+            molecularWeightStream << VectorStringBracketed(molWeights[1], true );
+            mostAbundantMassStream << VectorStringBracketed(mamMasses[1], true );
+            monoisotopicMassStream << VectorStringBracketed(misoMasses[1], true );
+            massCompositionStream << VectorStringBracketed(massCompositions[1], true );
+            grossFormulaStream << VectorStringBracketed(grossFormulas[1], true );
+        } else if( grossFormulas[0].size() || grossFormulas[1].size() )
+        {
+            int idx = grossFormulas[0].size() ? 0 : 1;
+            molecularWeightStream << VectorStringBracketed(molWeights[idx], false );
+            mostAbundantMassStream << VectorStringBracketed(mamMasses[idx], false );
+            monoisotopicMassStream << VectorStringBracketed(misoMasses[idx], false );
+            massCompositionStream << VectorStringBracketed(massCompositions[idx], false );
+            grossFormulaStream << VectorStringBracketed(grossFormulas[idx], false );
         }
```

Comment:
Don't we need to output something when both products and reactants values are empty?

Reply:
no, we don't need.